src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -2782,6 +2782,176 @@ export default class EthereumApi implements Api {
+              .toString()
+              .padStart(PAD_PRECISION, "0")}`
+          )
+        );
+      }
+
+      // baseFeePerGas
+      if (!baseFee) {
+        baseFeePerGas.unshift("0x0");
+      } else {
+        // Next block fee is based on existing blocks
+        if (currentBlockNumber === newestBlockNumber) {
+          baseFeePerGas.unshift(
+            Quantity.from(Block.calcNextBaseFee(currentBlock)).toString()
+          );
+        }
+        baseFeePerGas.unshift(Quantity.from(baseFee).toString());
+
+        // get reward percentile
+        if (rewardPercentiles) {
+          const transactions = currentBlock.getTransactions();
+
+          // If there are no transactions, all reward percentiles are 0.
+          if (transactions.length === 0) {
+            reward.unshift(
+              rewardPercentiles.map(() => {
+                return 0;
+              })
+            );
+          } else {
+            // Get receipts
+            const transactionReceipts = await Promise.all(
+              transactions.map(async (t, i) => {
+                return (
+                  await blockchain.transactionReceipts.get(t.hash.toString())
+                ).toJSON(currentBlock, transactions[i], blockchain.common);
+              })
+            );
+
+            // Get effective rewards and gas used for each transaction
+            const effectiveRewardAndGasUsed = transactionReceipts
+              .map((r, i) => {
+                const tx = transactions[i];
+                const baseFeePerGas =
+                  currentBlock.header.baseFeePerGas.toBigInt()
+                    ? currentBlock.header.baseFeePerGas.toBigInt()
+                    : ZERO_BIG_INT;
+
+                let effectiveGasReward;
+                if ("maxPriorityFeePerGas" in tx) {
+                  effectiveGasReward =
+                    tx.maxFeePerGas.toBigInt() - baseFeePerGas;
+                  if (tx.maxPriorityFeePerGas < effectiveGasReward) {
+                    effectiveGasReward = tx.maxPriorityFeePerGas;
+                  }
+                } else {
+                  effectiveGasReward = tx.gasPrice.toBigInt() - baseFeePerGas;
+                }
+
+                return {
+                  hash: tx.hash,
+                  effectiveGasReward,
+                  gasUsed: r.gasUsed ? r.gasUsed.toBigInt() : ZERO_BIG_INT
+                };
+              })
+              .sort((a, b) => {
+                if (a.effectiveGasReward > b.effectiveGasReward) return 1;
+                if (a.effectiveGasReward < b.effectiveGasReward) return -1;
+                return 0;
+              });
+
+            reward.unshift(
+              rewardPercentiles.map(p => {
+                let gasUsed = ZERO_BIG_INT;
+                // p can be a float
+                const targetGas =
+                  (currentBlock.header.gasUsed.toBigInt() *
+                    Quantity.from(p * PRECISION).toBigInt()) /
```

Comment:
I'd use `BigInt` directly
```suggestion
                    BigInt(p * PRECISION)) /
```